R/check_strata.R

```diff
@@ -0,0 +1,64 @@
+#' @title Check a `pop_data` object for requested strata variables
+#' @param pop_data a `pop_data` object
+#' @param strata a [character] vector
+#' @returns [NULL], invisibly
+#' @examples
+#' sees_pop_data_pk_100 |>
+#'   check_strata(strata = c("ag", "catch", "Count"))
+#' @dev
+check_strata <- function(pop_data, strata) {
+  if (!is.character(strata)) {
+    cli::cli_abort(
+      class = "strata are not strings",
+      message = c(
+        "x" = "Argument `strata` is not a character vector.",
+        "i" = "Provide a character vector with names of stratifying variables."
+      )
+    )
+  }
+
+  present_strata_vars <- intersect(strata, names(pop_data))
+  missing_strata_vars <- setdiff(strata, present_strata_vars)
+
+  if (length(missing_strata_vars) > 0) {
+    message0 <- c(
+      "Can't stratify provided {.arg pop_data}
+       with the provided {.arg strata}:",
+      "i" = "variable {.var {missing_strata_vars}}
+             {?is/are} missing in {.arg pop_data}."
+    )
+
+    partial_matches <-
+      purrr::map(missing_strata_vars, function(x) {
+        stringr::str_subset(string = names(pop_data), pattern = x) |>
+          glue::backtick() |>
```

Comment:
looks like these checks got removed from #227, but I added them back here and polished them up a bit

Reply:
That's weird.